src/h5web/visualizations/index.tsx

```diff
@@ -181,6 +181,12 @@ export const VIS_DEFS: Record<Vis, VisDef> = {
       );
 
       const dimsCount = dataset.shape.dims.length;
+      if (dimsCount === 0) {
+        throw new Error(
+          `Expected "${signal}" dataset to have at least one dimension`
+        );
+      }
```

Comment:
This check makes sense for `NxImage` as well, since a signal dataset with no dimension at all is wrong. The two `supportsEntity` function are now exactly identical except for the interpretation checks and for the dimension check with `NxImage`.